### ISSUE TITLE
Error Converting to CoreML: upsample_bicubic2d() received an invalid combination of arguments

### ISSUE THREAD

**Issue by kells1986** [facebookresearch/dinov2#102]
I want to run the small model on an iPhone. I have this code:

```python
import torch
import torchvision
import coremltools as ct

dinov2_vits14 = torch.hub.load('facebookresearch/dinov2', 'dinov2_vits14').eval()
with torch.no_grad():
    dinov2_vits14.requires_grad_(False)
    example_input = torch.rand(1, 3, 224, 224)
    traced_model = torch.jit.trace(dinov2_vits14, example_input) # fails here

mlmodel = ct.convert(
    dinov2_vits14,
    inputs=[ct.ImageType(name="input_1", shape=(1, 3, 224, 224), scale=1.0/255.0, bias=[0, 0, 0])],
    source="pytorch"
)
```

The code fails at the call to `torch.jit.trace` and gives the following error message:

```
TypeError: upsample_bicubic2d() received an invalid combination of arguments - got (Tensor, NoneType, bool, tuple), but expected one of:
 * (Tensor input, tuple of ints output_size, bool align_corners, tuple of floats scale_factors)
      didn't match because some of the arguments have invalid types: (Tensor, !NoneType!, bool, !tuple of (Tensor, Tensor)!)
 * (Tensor input, tuple of ints output_size, bool align_corners, float scales_h, float scales_w, *, Tensor out)
```

I'm running `torch==2.0.1` and `coremltools==6.1`. Any ideas how I can resolve this?

**Comment by TranThanh96** [comment]
same error when convert to onnx
```
TypeError: upsample_bicubic2d() received an invalid combination of arguments - got (Tensor, NoneType, bool, tuple), but expected one of:
 * (Tensor input, tuple of ints output_size, bool align_corners, tuple of floats scale_factors)
      didn't match because some of the arguments have invalid types: (Tensor, NoneType, bool, tuple)
 * (Tensor input, tuple of ints output_size, bool align_corners, float scales_h, float scales_w, *, Tensor out)

```

**Comment by DuongTSon** [comment]
The root cause of this issue is the onnx converts the scale_factor parameter in the interpolation function from float type to Tensor which is not correct.
To fix this issue, you can add a line of code to convert the Tensor to float type at the part that has the error in the `vision_transformer.py` file.

**Comment by patricklabatut** [comment]
Tracking requests for additional exports there: #167 instead.

**Comment by hardlipay** [comment]
> same error when convert to onnx
> 
> ```
> TypeError: upsample_bicubic2d() received an invalid combination of arguments - got (Tensor, NoneType, bool, tuple), but expected one of:
>  * (Tensor input, tuple of ints output_size, bool align_corners, tuple of floats scale_factors)
>       didn't match because some of the arguments have invalid types: (Tensor, NoneType, bool, tuple)
>  * (Tensor input, tuple of ints output_size, bool align_corners, float scales_h, float scales_w, *, Tensor out)
> ```

https://github.com/huggingface/transformers/blob/main/src/transformers/models/dinov2/modeling_dinov2.py#L107

I may have solved the problem by modifying the code here, but I don't know if it will cause other errors or performance loss. I've successfully converted to .onnx.

Here's the original code.

patch_pos_embed = nn.functional.interpolate(
            patch_pos_embed,
            scale_factor=(height / math.sqrt(num_positions), width / math.sqrt(num_positions)),
            mode="bicubic",
            align_corners=False,
        )
        
change to this:

desired_height = float(height / math.sqrt(num_positions))
desired_width = float(width / math.sqrt(num_positions))
patch_pos_embed = nn.functional.interpolate(
            patch_pos_embed,
            scale_factor=(desired_height , desired_width ),
            mode="bicubic",
            align_corners=False,
        )

**Comment by patricklabatut** [comment]
This should have been fixed with #247

**Comment by VimalMollyn** [comment]
@kells1986 I was able to convert the dinov2 models to coreml here: https://github.com/VimalMollyn/dinov2-coreml

**Comment by Norod** [comment]
> This should have been fixed with [#247](https://github.com/facebookresearch/dinov2/pull/247)

Just in case anyone Google's this and arrives here. To have DepthAnythingV2 (which uses dinov2) convert to CoreML with non  square 518x518 resolution, I used both the fix from this PR as well as changed to `mode="nearest"`

```
diff --git a/depth_anything_v2/dinov2.py b/depth_anything_v2/dinov2.py
index 83d2508..315ae07 100644
--- a/depth_anything_v2/dinov2.py
+++ b/depth_anything_v2/dinov2.py
@@ -190,20 +190,16 @@ class DinoVisionTransformer(nn.Module):
         h0 = h // self.patch_size
         # we add a small number to avoid floating point error in the interpolation
         # see discussion at https://github.com/facebookresearch/dino/issues/8
-        # DINOv2 with register modify the interpolate_offset from 0.1 to 0.0
-        w0, h0 = w0 + self.interpolate_offset, h0 + self.interpolate_offset
-        # w0, h0 = w0 + 0.1, h0 + 0.1
-        
+        w0, h0 = w0 + 0.1, h0 + 0.1
+
         sqrt_N = math.sqrt(N)
         sx, sy = float(w0) / sqrt_N, float(h0) / sqrt_N
         patch_pos_embed = nn.functional.interpolate(
             patch_pos_embed.reshape(1, int(sqrt_N), int(sqrt_N), dim).permute(0, 3, 1, 2),
             scale_factor=(sx, sy),
-            # (int(w0), int(h0)), # to solve the upsampling shape issue
-            mode="bicubic",
-            antialias=self.interpolate_antialias
+            mode="nearest",
         )
-        
+
         assert int(w0) == patch_pos_embed.shape[-2]
         assert int(h0) == patch_pos_embed.shape[-1]
         patch_pos_embed = patch_pos_embed.permute(0, 2, 3, 1).view(1, -1, dim)

```
